### PR TITLE
Enumerables

### PR DESCRIPTION
**Pull request by MarioDena**
Finished Main methods for the project.

It includes a test suite freely customizable.

### CODE REVIEW

File: enumerables.rb [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+
+module Enumerable
+  def my_each
+    i = 0
+    ar = self
+    while i < ar.size
+      yield(ar[i])
+      i += 1
+    end
+  end
```
[the remark this answers is not in the document]
Comment (MarioDena) [reply enumerables.rb]: Fixed

File: enumerables.rb [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+  def my_all?
+    i = 0
+    ar = self
+    while i < ar.size
+      return false unless yield(ar[i])
+
+      i += 1
+    end
+    true
+  end
```
[the remark this answers is not in the document]
Comment (MarioDena) [reply enumerables.rb]: Fixed

File: enumerables.rb [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+  def my_any?
+    i = 0
+    ar = self
+    while i < ar.size
+      return true if yield(ar[i])
+
+      i += 1
+    end
+    false
+  end
```
[the remark this answers is not in the document]
Comment (MarioDena) [reply enumerables.rb]: Fixed

File: enumerables.rb [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+  def my_none?
+    i = 0
+    ar = self
+    while i < ar.size
+      return false if yield (ar[i])
+
+      i += 1
+    end
+    true
+  end
```
[the remark this answers is not in the document]
Comment (MarioDena) [reply enumerables.rb]: Fixed

File: enumerables.rb [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+
+  def my_count(lit = '')
```
[the remark this answers is not in the document]
Comment (MarioDena) [reply enumerables.rb]: Fixed

File: enumerables.rb [hunk 6]
```diff
@@ -0,0 +1,137 @@
...
+    i = 0
+    ar = self
+    injector_result = origin
+    while i < ar.size
+      injector_result = yield(injector_result, ar[i])
+      i += 1
+    end
+    injector_result
+  end
+end
```
[the remark this answers is not in the document]
Comment (MarioDena) [reply enumerables.rb]: Fixed

File: enumerables.rb [hunk 7]
```diff
@@ -0,0 +1,137 @@
...
+      i += 1
+    end
+    true
+  end
+
+  def my_count(lit = '')
+    ar = self
+    i = 0
+    count = 0
+    if block_given? == false
```
[the remark this answers is not in the document]
Comment (MarioDena) [reply enumerables.rb]: I have no idea most of my code gets formatted by rubocop locally at first =)

**Comment by MarioDena** [comment]
Great review @bolah2009  
I've fixed all the issues and submitted a new code review.

Cheers!

File: enumerables.rb [hunk 8]
```diff
@@ -0,0 +1,207 @@
...
+p test_count
+p 'here'
+test_map = [1, 2, 3, 4, 5].my_map { |i| i * 4 }
+p test_map
+p [1, 2, 3, 4, 4, 5].my_inject(:+)
+p [1, 2, 3, 4, 4, 5].inject(:+)
+p multiply_els([1, 2, 3])
+
+# proc map method
+p [1, 2, 3, 4, 5].my_map(&my_proc)
```
Comment (stratospherique) [inline enumerables.rb]: - **[REQUIRED]** No debugging instructions in commited code (as discussed [here](https://questions.microverse.org/t/debugging-instructions-code/346))

File: enumerables.rb [hunk 9]
```diff
@@ -0,0 +1,207 @@
...
+      elsif args.empty? == true
+        return false unless ar[i]
+      else
+        return false unless args[i] == ar[i]
+      end
+
+      i += 1
+    end
+    true
+  end
```
Comment (stratospherique) [inline enumerables.rb]: * [ ] enumerables my_all? when a class is passed as an argument returns true if all of the collection is a member of such class
  ```
    array = [1,2,5,4,7]
    p array.my_all?(Integer) == array.all?(Integer) # false
  ```
* [ ] enumerables my_all? when a Regex is passed as an argument returns true if all of the collection matches the Regex
  ```
    array = %w[dog door rod blade]
    p array.my_all?(/d/) == array.all?(/d/) # false
  ```
* [ ] enumerables my_all? when no block or argument is given returns true when none of the collection members are false or nil
  ```
    true_array = [1, true, 'hi', []]
    p true_array.my_all? == true_array.all? # false
  ```
* [ ] enumerables my_all? when a pattern other than Regex or a Class is given returns true if all of the collection matches the pattern
  ```
    array = [3,3,3,3,3]
    p array.my_all?(3) == array.all?(3) # false
  ```

File: enumerables.rb [hunk 10]
```diff
@@ -0,0 +1,207 @@
...
+
+  def my_any?(*args)
+    i = 0
```
Comment (stratospherique) [inline enumerables.rb]: - Same Issues as my_all?.

File: enumerables.rb [hunk 11]
```diff
@@ -0,0 +1,207 @@
...
+      end
+
+      i += 1
+    end
+    false
+  end
+
+  def my_none?(*args)
+    i = 0
+    ar = self
```
Comment (stratospherique) [inline enumerables.rb]: - Same Issues as my_all?.

File: enumerables.rb [hunk 12]
```diff
@@ -0,0 +1,207 @@
...
+                          injector_result.send(symbol, ar[i])
+                        else
+                          yield(injector_result, ar[i])
+                        end
+      i += 1
+    end
+
+    injector_result
+  end
+end
```
Comment (stratospherique) [inline enumerables.rb]: * [ ] enumerables my_inject combines all elements of enum by applying a binary operation, specified by a block
  ```
    range = Range.new(5,50)
    p range.my_inject(4) { |prod, n| prod * n } 
    # return this errors 
    NoMethodError:
        undefined method `[]' for 5..50:Range
      # ./enumerables.rb:165:in `my_inject'
  ```


* [ ] enumerables my_inject when a symbol is specified combines each element of the collection by applying the symbol as a named method
  ```
    range = Range.new(5,50)
    p range.my_inject(2, :*) 
    # return this errors 
    NoMethodError:
        undefined method `[]' for 5..50:Range
      # ./enumerables.rb:163:in `my_inject'

  ```

File: enumerables.rb (hunk 8 above)
Comment (MarioDena) [reply enumerables.rb]: Those are not debugging instructions, but a test suite. The difference can be found here: https://www.geeksforgeeks.org/differences-between-testing-and-debugging/

File: enumerables.rb (hunk 9 above)
Comment (MarioDena) [reply enumerables.rb]: Fixed

File: enumerables.rb (hunk 10 above)
Comment (MarioDena) [reply enumerables.rb]: fixed

File: enumerables.rb (hunk 11 above)
Comment (MarioDena) [reply enumerables.rb]: fixed

File: enumerables.rb (hunk 12 above)
Comment (MarioDena) [reply enumerables.rb]: fixed

**Comment by MarioDena** [comment]
@stratospherique  
Fixed all instances, except for 
"[REQUIRED] No debugging instructions in committed code (as discussed here)" As I didn't include debugging instructions please refer to my comment.

Changed the target branch for merge to "dev" as to follow git flow

New code request submitted.

Cheers!

**Pull request merged**